app/repository/RaceRepositoryImpl.java

```diff
@@ -23,35 +29,80 @@ public RaceRepositoryImpl(JPAApi jpaApi, DatabaseExecutionContext databaseExecut
     }
 
     @Override
-    public CompletionStage<Race> getRace() {
-        return supplyAsync(() -> wrap (this::getRace), databaseExecutionContext);
+    public CompletionStage<JsonNode> getAllRaces() {
+        return supplyAsync(() -> wrap (this::getAllRaces), databaseExecutionContext);
     }
 
-    private Race getRace(EntityManager em) {
+    private JsonNode getAllRaces(EntityManager em) {
         List<Race> races = em.createQuery("select r from Race r", Race.class).getResultList();
-        return races.get(0);
+        return toJson(races);
     }
 
     @Override
-    public void setRace(Race race) {
-        deleteAllRaces();
-        jpaApi.em().getTransaction().begin();
-        jpaApi.em().persist(race);
-        jpaApi.em().getTransaction().commit();
+    public CompletionStage<JsonNode> getRace(int raceId) {
+        return supplyAsync(() -> wrap (em -> getRace(em, raceId)), databaseExecutionContext);
+    }
+
+    private JsonNode getRace(EntityManager em, int raceId) {
+        TypedQuery<Race> query = em.createQuery("select r from Race r where r.raceId = :raceId" , Race.class);
+        query.setParameter("raceId", raceId);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "raceId" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)